src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -568,6 +644,91 @@ export default class AircraftController {
     }
 
     /**
+     * Generate a unique `transponderCode`
+     *
+     * This method should only be run while building props for a
+     * soon-to-be-instantiated `AircraftModel`
+     *
+     * @for AircraftController
+     * @method _generateUniqueTransponderCode
+     * @return {number}
+     * @private
+     */
+    _generateUniqueTransponderCode() {
+        const transponderCode = generateRandomOctalWithLength(4);
+
+        if (!this._isDiscreteTransponderCode(transponderCode)) {
+            // the value generated is already in use, recurse back through this method and try again
+            this._generateUniqueTransponderCode();
+        }
+
+        this._addTransponderCodeToInUse(transponderCode);
+
+        return transponderCode;
+    }
+
+    /**
+     * Add a given `transponderCode` to the `#_transponderCodesInUse` list
+     *
+     * @for AircraftController
+     * @method _addTransponderCodeToInUse
+     * @param transponderCode {number}
+     */
+    _addTransponderCodeToInUse(transponderCode) {
+        this._transponderCodesInUse.push(transponderCode);
+    }
+
+    /**
+     * Remove the `transponderCode` from the list of `#_transponderCodesInUse`
+     *
+     * @for AircraftController
+     * @method _removeTransponderCodeFromUse
+     * @param transponderCode {number}
+     */
+    _removeTransponderCodeFromUse({ transponderCode }) {
+        this._transponderCodesInUse = _without(this._transponderCodesInUse, transponderCode);
+    }
+
+    /**
+     * Boolean helper used to determine if a given `transponderCode` is already
+     * present within the `#_transponderCodesInUse` list.
+     *
+     * @for AircraftController
+     * @method _isTransponderCodeInUse
+     * @param transponderCode {number}
+     * @return {booelean}
+     */
+    _isTransponderCodeInUse(transponderCode) {
+        return this._transponderCodesInUse.indexOf(transponderCode) !== -1;
+    }
+
+    /**
+     * Boolean helper used to determine if a given `transponderCode` is both
+     * not in use and not an invalid transponderCode.
+     *
+     * @for AircraftController
+     * @method _isValidTransponderCode
+     * @param transponderCode {number}
+     * @return {boolean}
+     */
+    _isValidTransponderCode(transponderCode) {
+        return RESERVED_SQUAWK_CODES.indexOf(transponderCode) === -1;
+    }
+
+    /**
+     * Helper used to determine if a given `transponderCode` is both
+     * valid and not in use.
+     *
+     * @for AircraftController
+     * @method _isDiscreteTransponderCode
+     * @param transponderCode {number}
+     * @return {boolean}
+     */
+    _isDiscreteTransponderCode(transponderCode) {
+        return !this._isTransponderCodeInUse(transponderCode) && this._isValidTransponderCode(transponderCode);
+    }
```

Comment:
In order to be "discrete", it must be valid and not one of the reserved codes. It being in use or not in use is not related to it being discrete.

`return this._isValidTransponderCode(transponderCode) && RESERVED_SQUAWK_CODES.indexOf(transponderCode) === -1;`